### PR TITLE
re-implement AbortController PR w/latest version

### PR DESCRIPTION
**Pull request by logandeancall**
Re-implement the 2018 PR with AbortController but based on the most current version. Also, implement suggestions from maintainer in that PR.

### CODE REVIEW

File: src/Fetch.ml [hunk 1]
```diff
@@ -14,6 +17,14 @@ type formData (* XMLHttpRequest *)
 type readableStream (* Streams *)
 type urlSearchParams (* URL *)
 
+module AbortController = struct
+  type t = abortController
+
+  external signal : t -> signal = "signal" [@@bs.get]
+  external abort : t -> unit = "abort" [@@bs.send.pipe: t]
```
Comment (glennsl) [inline src/Fetch.ml]: The `t` argument needs to be removed here. It's implied by the `t` passed to `@bs.send.pipe`.

File: src/Fetch.mli [hunk 2]
```diff
@@ -26,6 +28,18 @@ type requestMethod =
...
   | Other of string
 
+module AbortController : sig
+(* Experimental API *)
+  type t = abortController
+
+(* Experimental API *)
+  external signal : t -> signal = "signal" [@@bs.get]
+(* Experimental API *)
+  external abort : t -> unit = "abort" [@@bs.send]
```
Comment (glennsl) [inline src/Fetch.mli]: This doesn't match the implementation.

**Review by glennsl — CHANGES_REQUESTED** [review]
Thanks! Unfortunately this doesn't actually compile. See the comments below. But once those have been fixed, I think this should be good to go.

File: src/Fetch.ml (hunk 1 above)
Comment (logandeancall) [reply src/Fetch.ml]: Okay, sorry I'm only familiar with Reason syntax and even then some of this still alludes me. So I'm assuming the line should be changed to this?
```reason
external abort : unit -> unit = "abort" [@@bs.send.pipe: t]
```

Comment (glennsl) [reply src/Fetch.ml]: No, it should be just `external abort : unit = "abort" [@@bs.send.pipe: t]`.

I have [a handy cheatsheet](https://bucklescript.github.io/bucklescript/Manual.html#_binding_to_method_code_bs_send_code_code_bs_send_pipe_code) you can refer to, which might help.

File: src/Fetch.ml [hunk 3]
```diff
@@ -14,6 +17,14 @@ type formData (* XMLHttpRequest *)
 type readableStream (* Streams *)
 type urlSearchParams (* URL *)
 
+module AbortController = struct
+  type t = abortController
+
+  external signal : t -> signal = "signal" [@@bs.get]
+  external abort : unit -> unit = "abort" [@@bs.send.pipe: t]
```
Comment (glennsl) [inline src/Fetch.ml]:
```suggestion
  external abort : unit = "abort" [@@bs.send.pipe: t]
```

File: src/Fetch.mli [hunk 4]
```diff
@@ -26,6 +28,18 @@ type requestMethod =
...
   | Other of string
 
+module AbortController : sig
+(* Experimental API *)
+  type t = abortController
+
+(* Experimental API *)
+  external signal : t -> signal = "signal" [@@bs.get]
+(* Experimental API *)
+  external abort : unit -> unit = "abort" [@@bs.send.pipe: t]
```
Comment (glennsl) [inline src/Fetch.mli]:
```suggestion
  external abort : unit = "abort" [@@bs.send.pipe: t]
```

**Review by glennsl — APPROVED** [review]
I forgot I can suggest and approve my own changes, so fixed! Thanks for the contribution!

**Pull request merged**